Framework/Core/src/ResourcesMonitoringManager.cxx

```diff
@@ -0,0 +1,110 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "ResourcesMonitoringManager.h"
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <fstream>
+#include <string_view>
+#include <algorithm>
+#include <sstream>
+
+
+using namespace o2::framework;
+
+template <class... Fs> struct ResourcesMonitoringOverload : Fs... { using Fs::operator()...; };
+template <class... Fs> ResourcesMonitoringOverload(Fs...) -> ResourcesMonitoringOverload<Fs...>;
+
+void ResourcesMonitoringManager::addNewMetrics(const std::string& specName, 
+    std::vector<PerformanceMetric>&& parsedMetrics) noexcept
+{
+    for(auto&& metric : parsedMetrics)
+    {
+        m_data[specName].push_back(metric);
+    }
+}
+
+
+bool ResourcesMonitoringManager::dumpMetricsToJSON() noexcept
+{
+    
+    if(m_data.empty()){ return false; }
+
+    boost::property_tree::ptree root;
+    for (auto& [device, metrics] : m_data)
+    {
+        
+        boost::property_tree::ptree deviceRoot;
+        std::unordered_map<std::string, boost::property_tree::ptree> metricNodes; 
+
+        for(auto& metric : metrics)
+        {
+            boost::property_tree::ptree values;
+            values.add("timestamp", metric.m_timestamp);
+            values.add("value", metric.m_value);
+            metricNodes[metric.m_name].push_back(std::make_pair("", values));
+        }
+
+        for(auto& [metricName, metricTree] : metricNodes)
+        {
+            deviceRoot.add_child(metricName, metricTree);
+        }
+        root.add_child(device, deviceRoot);
+    }
+
+    //to be discussed, should be constant file name or provided as parameter
+    std::ofstream file("resourcesMetrics.json", std::ios::out);
+    if(file.is_open()) { boost::property_tree::json_parser::write_json(file, root); }
+    else { return false; }
+
+    file.close();
+    
+    return true;
+
+}
+
+
+std::optional< std::vector<PerformanceMetric> > ResourcesMonitoringHelper::parsePerfomanceMetric(
+    const ResourcesMonitoringManager& manager, const std::string& line)
+{
+    if(! manager.isResourcesMonitoringEnabled()){ return {}; }
+
+    // [METRIC] <performanceMetricName> metric1=val1 metric2=val2 .... hostname=<hostname>    
+    static const std::string performanceMetricPrefix = "[METRIC] " + monitoring::ProcessMonitor::getPerformanceMetricName();
+    if(line.length() < performanceMetricPrefix.length()) { return {}; }
+    
+    if(auto res = std::mismatch(performanceMetricPrefix.begin(), performanceMetricPrefix.end(), 
+        line.begin()); res.first != performanceMetricPrefix.end())
+    { return {}; }
+
+    std::istringstream iss(line);
+    std::vector<std::string> elements(std::istream_iterator<std::string>{iss}, std::istream_iterator<std::string>());
+
+    std::vector<PerformanceMetric> result(elements.size() - 4); // -1 prefix -1 main metric name -1 timestamp -1 hostname
```

Comment:
The current parser is allocation-less. Please make sure it remains such. Why do you need a new one? Can't you adapt the old one?

Reply:
The old parser works only for convention -> single metric - single value with type. Adam suggested to change it to -> single metric - multiple value which cannot be parsed with existing parser. I will change the new one to be allocation-less.